### PR TITLE
client: add support for "ResumeToken", "HeaderPeek" to download

### PR DESCRIPTION
**Pull request by mvo5**
This adds the client bits for the improved download API as
developed in PR #8078.

### CODE REVIEW

File: client/snap_op.go [hunk 1]
```diff
@@ -314,20 +314,31 @@ type snapRevisionOptions struct {
...
 	SuggestedFileName string
 	Size              int64
 	Sha3_384          string
+	ResumeToken       string
+}
+
+type DownloadOptions struct {
+	SnapOptions
+
+	HeaderPeek  bool   `json:"header-peek,omitempty"`
```
Comment (pedronis) [inline client/snap_op.go]: this doesn't look like it is sent to the daemon directly, so not sure we need the json tags

File: client/snap_op.go [hunk 2]
```diff
@@ -314,20 +314,31 @@ type snapRevisionOptions struct {
...
+	ResumeToken string `json:"resume-token,omitempty"`
+
 	snapRevisionOptions
 }
 
 type DownloadInfo struct {
 	SuggestedFileName string
 	Size              int64
 	Sha3_384          string
+	ResumeToken       string
```
Comment (pedronis) [inline client/snap_op.go]: this addition is not tested afaict

File: client/snap_op.go [hunk 3]
```diff
@@ -314,20 +314,31 @@ type snapRevisionOptions struct {
...
+}
+
+type DownloadOptions struct {
+	SnapOptions
+
+	HeaderPeek  bool
+	ResumeToken stringgit
```
Comment (pedronis) [inline client/snap_op.go]: stringgit :)

File: client/snap_op.go [hunk 4]
```diff
@@ -314,20 +314,31 @@ type snapRevisionOptions struct {
 }
 
 type downloadAction struct {
-	SnapName string `json:"snap-name,omitempty"`
+	SnapName    string `json:"snap-name"`
+	HeaderPeek  bool   `json:"header-peek,omitempty"`
```
Comment (pedronis) [inline client/snap_op.go]: I would put these two new fields at the end of struct:

```
type ... {
   SnapName
   
   snapRevisionOptions

   HeaderPeek
   ResumeToken
}
```

File: client/snap_op.go (hunk 3 above)
Comment (mvo5) [reply client/snap_op.go]: Yes, sorry for that

Comment (zyga) [reply client/snap_op.go]: We need a T-shift for this :D

**Comment by mvo5** [comment]
@zyga Yeah, there is more to come - the bits that actually use this code in the snap download command.

**Pull request merged**